### PR TITLE
explain and use the `brooklyn-demo-sample` external provider

### PR DESCRIPTION
**Pull request by ahgittin**
so far just explains it, need to use it

### CODE REVIEW

**Comment by ahgittin** [comment]
examples all updated, ready to merge after https://github.com/apache/brooklyn-server/pull/812

**Comment by ahgittin** [comment]
@drigodwin @aledsage merged the upstream PRs -- is this good to go?

(and if you could have a quick look at #208 i'd be obliged!)

File: guide/blueprints/multiple-services.md [hunk 1]
```diff
@@ -41,8 +41,11 @@ Here there are a few things going on:
 * We've added a second service, which will be the database;
   you'll note the database has been configured to run a custom setup script
 * We've injected the URL of the second service into the appserver as a Java system property
-  (so our app knows where to find the database) 
-
+  (so our app knows where to find the database)
+* We've used externalized config to keep secret information out of the blueprint;
+  this is loaded at runtime from an [externalized config provider](../ops/externalized-config.html),
+  such as a remote credentials store
```
Comment (drigodwin) [inline guide/blueprints/multiple-services.md]: Missing full stop.

File: guide/ops/cli/cli-usage-guide.md [hunk 2]
```diff
@@ -439,7 +439,7 @@ tvZoNUTN   ssh: launching NginxControllerImpl{id...   Sun Dec 20 19:18:08 GMT 20
 {% endhighlight %}
 
 ## YAML Blueprint
-This the YAML blueprint used for this document.
+This the YAML blueprint used for this document, based on the [web cluster](../../blueprints/clusters-and-policies.html) examples.
```
Comment (drigodwin) [inline guide/ops/cli/cli-usage-guide.md]: This the -> This is the

File: guide/ops/externalized-configuration.md [hunk 3]
```diff
@@ -64,6 +68,18 @@ brooklyn.external.credentials.appId=MyApp
 
 In this case, the list of servers is stored in a properties file located on an Operations Department web server, and the
 credentials are stored in an instance of [Vault](https://www.vaultproject.io/).
+More information on these providers is below.
+
+For demo purposes, there is a pre-defined external provider called
+`brooklyn-demo-sample` which defines `hidden-brooklyn-password` as `br00k11n`.
+This is used in some of the sample blueprints, referencing `$brooklyn:external("brooklyn-demo-sample", "hidden-brooklyn-password")`. 
+The value used here can be overridden with the following in your `brooklyn.properties`:
```
Comment (drigodwin) [inline guide/ops/externalized-configuration.md]: `brooklyn.cfg` as the next release will be karaf?

File: guide/blueprints/multiple-services.md (hunk 1 above)
Comment (ahgittin) [reply guide/blueprints/multiple-services.md]: style issue - omitting full stops on single-sentence bullet points is common, don't feel strongly but should ideally be conisstent w two bullet poitns above (and elsewhere), which it is

File: guide/ops/externalized-configuration.md (hunk 3 above)
Comment (ahgittin) [reply guide/ops/externalized-configuration.md]: that will need to be changed in lots of places, including the first para of this doc.  changing just here is weird.  suggest it be a different tranch of work.

**Comment by ahgittin** [comment]
updated.  suggest this be backported to 0.12.0 in addition to master.

File: guide/ops/externalized-configuration.md (hunk 3 above)
Comment (drigodwin) [reply guide/ops/externalized-configuration.md]: They have been changed in #210 already. I'll merge this though and make a new PR.

**Pull request merged**